## Задача №1 - NotFoundException

### Легенда

Вы развиваете приложение с менеджером товаров, который мы рассматривали на лекции, и решили сделать так, чтобы при попытке удаления несуществующего объекта генерировалось ваше исключение, а не `ArrayIndexOfBoundsException`.

Обратите внимание: это правильный подход, поскольку таким образом вы сообщаете (через генерацию исключения), что это исключение, вписывающееся в вашу логику, а не ошибка программиста.

**Исключения являются частью API**.

Что нужно сделать:
1. Возьмите проект (мы его писали на одной из предыдущих лекций) с менеджером, репозиторием и товарами (`Product`, `Book`, `TShirt`)
1. Создайте класс исключения `NotFoundException` отнаследовавшись от `RuntimeException` и реализуйте все конструкторы
1. В методе удаления `removeById` сначала проверяйте, есть ли элемент (для этого прямо из метода `removeById` вызывайте метод `findById`: если результат - `null`, тогда выкидывайте исключение `NotFoundException`*)
1. Напишите 2 автотеста на репозиторий: первый должен проверять успешность удаления существующего элемента, второй - генерации `NotFoundException` при попытке удаления несуществующего элемента

Убедитесь, что ваши автотесты проходят (напоминаем, что проект должен быть на базе Maven, с подключенными зависимостями и необходимыми плагинами).

Итого: у вас должен быть репозиторий на GitHub, в котором расположен ваш Java-код и автотесты к нему.

Примечание*: обратите внимание, когда генерируются стандартные исключения, всегда передаётся сообщение (на английском языке), например:

![](pic/exception.png)

Мы рекомендуем вам делать так же и указывать в сообщении: при удалении по какому конкретно id было сгенерировано ваше исключение.

Сделать это можно следующим образом (простейший способ): ```"Element with id: " + id + " not found"```.
